src/planner.c

```diff
@@ -528,29 +505,59 @@ timescaledb_set_rel_pathlist(PlannerInfo *root,
 							 RangeTblEntry *rte)
 {
 	Hypertable *ht;
-
-	global_planner_context_init();
+	Cache	   *hcache;
 
 	if (prev_set_rel_pathlist_hook != NULL)
 		(*prev_set_rel_pathlist_hook) (root, rel, rti, rte);
 
-	if (!extension_is_loaded() ||
-		IS_DUMMY_REL(rel) ||
-		!should_optimize_query() ||
-		!OidIsValid(rte->relid))
-		goto out_release;
+	if (!extension_is_loaded() || IS_DUMMY_REL(rel) || !OidIsValid(rte->relid))
+		return;
 
-	sort_transform_optimization(root, rel);
+	hcache = hypertable_cache_pin();
+	ht = hypertable_cache_get_entry(hcache, rte->relid);
```

Comment:
This lookup should probably move to after `RELOPT_OTHER_MEMBER` check (see comment below).